### PR TITLE
Better errors and versions

### PR DESCRIPTION
**Pull request by barbariand**
This pull request enhances error handling and parser orginsiation.

**Changes:**

* **Improved Error Handling:**  Previously the errors where all &'static str and is now a `NodeParseError`.
* **New Crate:** Moved the parsing to a new crate.
* **New Type for Byte Tracking:**  Instead of using `Iterator<Item=u8>`, a new type `Lexer<T:Iteartor<Item=u8>>` is introduced. This wrapper keeps track of the number of bytes read witch makes errors clearer in files.
* **Added basic arguments support** Clap allows us to parse input flags for if we want to have side image generation or smth.
* **Start of versioned Node definitions** Added a new type `Node` that has all the versions and is the one responsible for parsing out the version.

**Benefits:**

* More robust and reliable error handling.
* Enhanced error reporting with precise byte information.
* A `Node` type can maybe help us i still do not know how we should manage versioning in the future

### CODE REVIEW

File: output
Comment (JoshuaCF) [inline output:1]: Not sure you meant to include this one, since this is just a debug file really.

Comment (barbariand) [reply output:1]: Yea i did not mean to include it. Removed :)

**Comment by JoshuaCF** [comment]
No idea how to really handle versioning either, tbh right now I was just ignoring the issue and I was going to only maintain one that works on the latest version. I should probably do some research on how other software handles this so we can get some ideas.

File: kronark-node-parser/src/errors.rs [hunk 1]
```diff
@@ -0,0 +1,37 @@
+use std::error::Error;
+use std::fmt::Display;
+use std::string::FromUtf8Error;
+#[derive(Debug)]
+pub enum NodeParseError {
+    EOF(&'static str, u64),
+    InvalidFile(u64),
+    InvalidVersion(u64),
+    UTF8EncodingError(FromUtf8Error, u64),
+    FileToLong(u64),
```
Comment (JoshuaCF) [inline kronark-node-parser/src/errors.rs]: Typo: `FileTooLong` is what I believe you meant

**Review by JoshuaCF — APPROVED** [review]

**Review by JoshuaCF — APPROVED** [review]

**Pull request merged**